cmd/extproc/mainlib/main.go

```diff
@@ -116,10 +121,21 @@ func Main() {
 	s := grpc.NewServer()
 	extprocv3.RegisterExternalProcessorServer(s, server)
 	grpc_health_v1.RegisterHealthServer(s, server)
+
+	// Start metrics server.
+	metricsServer := startMetricsServer(flags.metricsAddr, l)
+
 	go func() {
 		<-ctx.Done()
 		s.GracefulStop()
+
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
not sure what is this for. why 5 sec and close? can you just remove and use the ctx (signal handle already is handled) and pass to Shutdown below.

Reply:
The `ctx` is already done (a couple lines above), so probably we shouldn't use it here? I don't know if we need this timeout, but the idea is to not use the already completed `ctx.

Reply:
oh yeah